rmw_connext_cpp/src/functions.cpp

```diff
@@ -164,62 +212,73 @@ rmw_create_publisher(
     rmw_set_error_string("callbacks handle is null");
     return NULL;
   }
-  std::string type_name = std::string(callbacks->package_name) + "::msg::dds_::" +
-    callbacks->message_name + "_";
+  std::string type_name = _create_type_name(callbacks, "msg");
+  // Past this point, a failure results in unrolling code in the goto fail block.
```

Comment:
Ok, I can look at prefixing the function name to the label (lots of busy work), but honestly when ever you encounter a `goto fail;` the very next instance of `fail:` should be the label it goes to and so searching for it should be straight forward.